src/main/java/com/hillel/dzhurynskyi/homework/hm33/Cart.java

```diff
@@ -0,0 +1,42 @@
+package com.hillel.dzhurynskyi.homework.hm33;
+
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE, proxyMode = ScopedProxyMode.DEFAULT)
+public class Cart {
+
+    private final List<Product> cart = new ArrayList<>();
+
+
+    public void addProductById(int id, ProductRepository productRepository) {
```

Comment:
private final ProductRepository productRepository. And constructor for it (for autowiring)